### PR TITLE
CEL validation: implement validateProvider for EnvoyProxy

### PR DESCRIPTION
**Pull request by zirain**

### CODE REVIEW

File: api/v1alpha1/envoyproxy_types.go [hunk 1]
```diff
@@ -122,6 +122,9 @@ type EnvoyProxyKubernetesProvider struct {
 	// are applied.
 	//
 	// +optional
+	// +kubebuilder:validation:XValidation:message="allocateLoadBalancerNodePorts can only be set for LoadBalancer type",rule="!has(self.allocateLoadBalancerNodePorts) || self.type == 'LoadBalancer'"
+	// +kubebuilder:validation:XValidation:message="loadBalancerIP can only be set for LoadBalancer type",rule="!has(self.loadBalancerIP) || self.type == 'LoadBalancer'"
+	// +kubebuilder:validation:XValidation:message="loadBalancerIP must be a valid IPv4 address",rule="!has(self.loadBalancerIP) || self.loadBalancerIP.matches(r\"^(((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\.|$)){4})\")"
```
Comment (zirain) [inline api/v1alpha1/envoyproxy_types.go:127]: this's hard to read, but I cannot find better solution right now.
Hope one day, there's ip lib from kubernetest just like [url](https://kubernetes.io/docs/reference/using-api/cel/#kubernetes-url-library).

File: tools/make/kube.mk [hunk 2]
```diff
@@ -49,7 +51,7 @@ kube-generate: $(tools/controller-gen) ## Generate code containing DeepCopy, Dee
 kube-test: manifests generate $(tools/setup-envtest) ## Run Kubernetes provider tests.
 	@$(LOG_TARGET)
 	KUBEBUILDER_ASSETS="$(shell $(tools/setup-envtest) use $(ENVTEST_K8S_VERSION) -p path)" go test --tags=integration ./... -coverprofile cover.out
-	KUBEBUILDER_ASSETS="$(shell $(tools/setup-envtest) use $(ENVTEST_K8S_VERSION) -p path)" go test --tags=celvalidation ./... -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell $(tools/setup-envtest) use $(CEL_VALIDATION_K8S_VERSION) -p path)" go test --tags=celvalidation ./test/... -coverprofile cover.out
```
[the remark this answers is not in the document]
Comment (zirain) [reply tools/make/kube.mk:54]: sorry, I didn't got your point.
`kube-test` is part of `go.test.coverage`

[the remark this answers is not in the document]
Comment (arkodg) [reply tools/make/kube.mk:54]: `kube-test` is **not** part of `go.test.coverage` , it runs `go test` with tags directly

File: api/v1alpha1/envoyproxy_types.go (hunk 1 above)
Comment (AliceProxy) [reply api/v1alpha1/envoyproxy_types.go:127]: Unfortunately, this is just kind of the nature of CEL for more complex logic.

**Review by AliceProxy — APPROVED** [review]

File: test/cel-validation/envoyproxy_test.go [hunk 3]
```diff
@@ -3,10 +3,10 @@
 // The full text of the Apache license is available in the LICENSE file at
 // the root of the repo.
 
-//go:build validation
```
Comment (arkodg) [inline test/cel-validation/envoyproxy_test.go:6]: can we rename file to
```
test/cel/envoyproxy_test.go
```

and the tag can be `celvalidation`

File: tools/make/kube.mk [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION ?= 1.24.1
+# CEL_VALIDATION_K8S_VERSION refers to the version of kubebuilder assets to be downloaded in validation test.
```
Comment (arkodg) [inline tools/make/kube.mk:3]: 🙈 
please raise a GH issue to track this work around

Comment (zirain) [reply tools/make/kube.mk]: it's fixed, hope CI is happy.

File: tools/make/kube.mk [hunk 5]
```diff
@@ -49,7 +49,6 @@ kube-generate: $(tools/controller-gen) ## Generate code containing DeepCopy, Dee
 kube-test: manifests generate $(tools/setup-envtest) ## Run Kubernetes provider tests.
 	@$(LOG_TARGET)
```
Comment (arkodg) [inline tools/make/kube.mk:50]: can you also update L51 to include `celvalidation` tag

Comment (zirain) [reply tools/make/kube.mk:50]: done

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**